tests/integration/optlyplugins/track_listener_manager.go

```diff
@@ -0,0 +1,61 @@
+/****************************************************************************
+ * Copyright 2019, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+package optlyplugins
+
+import (
+	"github.com/optimizely/go-sdk/pkg/entities"
+	"github.com/optimizely/go-sdk/pkg/event"
+	"github.com/optimizely/go-sdk/tests/integration/models"
+)
+
+// TrackListenerManager helper for track notification
+type TrackListenerManager struct {
+	listenersCalled []models.TrackListener
+}
+
+// TrackCallback represents callback function for track notification
+type TrackCallback = func(eventKey string, userContext entities.UserContext, eventTags map[string]interface{}, conversionEvent event.ConversionEvent)
```

Comment:
am not in favor of adding more and more new `types` for every listener, I think only one `interface` is fineIf there is any issue lets discuss offline.